### ISSUE TITLE
cfn-lint treats dynamic refs in AWS::CloudFormation::Stack TemplateURL as string literals

### ISSUE THREAD

**Issue by cnfait** [aws-cloudformation/cfn-lint#4151]
### CloudFormation Lint Version

1.35.1

### What operating system are you using?

Debian/Linux

### Describe the bug

cfn-lint doesn't resolve dynamic references (provided with `{{resolve:}}`, such as ` TemplateURL: "{{resolve:ssm:/sdlf/pipeline/main:1}}"`) in `AWS::CloudFormation::Stack` `TemplateURL` properties.

It tries to use them as string literals, directly part of the file location:
```
2025-05-22 12:38:01,938 - cfnlint.decode.decode - ERROR - Template file not found: /home/example/data-lakes-on-aws/sdlf-stage-lambda/src/{{resolve:ssm:/sdlf/pipeline/main:1}}
```

### Expected behavior

Ideally I'd like these values to be resolved and the proper values to be used, although this might prove challenging/out of scope for cfn-lint. 

Perhaps more realistically, not trying to use the dynamic reference as a raw string, and printing an informational-level message would be good enough for my use case.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Transform: AWS::Serverless-2016-10-31
Description: Example

Parameters:
  pPipelineReference:
    Description:  example
    Type: String
    Default: none

Resources:
  rPipelineInterface:
    Type: AWS::CloudFormation::Stack
    DeletionPolicy: Delete
    UpdateReplacePolicy: Delete
    Properties:
      TemplateURL: "{{resolve:ssm:/sdlf/pipeline/main:1}}"
      Parameters:
        pPipelineReference: !Ref pPipelineReference
```

**Comment by kddejong** [comment]
The exit code for this is still `0` but we do log the error into stderr.  We also don't make calls into a AWS account so we have a few options left.

* Suppress the logger in rule `E3043`
* See if the TemplateURL is a dynamic reference and skip it.

I think the easiest would be option 2.
